### PR TITLE
Add the Django debug toolbar to the root Poetry group

### PR DESCRIPTION
**Pull request by ahosgood**

### CODE REVIEW

File: config/urls.py [hunk 1]
```diff
@@ -15,14 +15,22 @@
...
+        path("", include(("app.main.urls", "main"), namespace="main")),
+        path("healthcheck/", include("app.healthcheck.urls")),
+        path(
+            "search/",
+            include(("app.search.urls", "search"), namespace="search"),
+        ),
+        path("admin/", admin.site.urls),
+    ]
+    + static(settings.STATIC_URL, document_root=settings.STATIC_ROOT)
+    + debug_toolbar_urls()
```
Comment (TNA-Allan) [inline config/urls.py:35]: Could this `debug_toolbar_urls()` fail  for a non root user? 

Could this be added only if installed? as in 

`from django.apps import apps`

```
if apps.is_installed("debug_toolbar"):
    urlpatterns = [
        path("__debug__/", include("debug_toolbar.urls")),
    ] + urlpatterns
```

File: config/settings/develop.py [hunk 2]
```diff
@@ -7,6 +7,14 @@
...
 ALLOWED_HOSTS = os.environ.get("ALLOWED_HOSTS", "*").split(",")
 
+INSTALLED_APPS = INSTALLED_APPS + [  # noqa: F405
+    "debug_toolbar",
+]
+
+MIDDLEWARE = [
+    "debug_toolbar.middleware.DebugToolbarMiddleware",
+] + MIDDLEWARE  # noqa: F405
+
```
Comment (TNA-Allan) [inline config/settings/develop.py:17]: As DEBUG setting is env variable driven. Could this be moved under DEBUG setting/end of page with a condition setting?

```
if DEBUG :
    from .base import INSTALLED_APPS, MIDDLEWARE

    INSTALLED_APPS += [
        "debug_toolbar",
    ]

    MIDDLEWARE += [
        "debug_toolbar.middleware.DebugToolbarMiddleware",
    ]
```

Comment (TNA-Allan) [reply config/settings/develop.py:17]: Additionally add in the condition - this was used to show the toolbar.

```
def show_toolbar(request) -> bool:
        return True

    DEBUG_TOOLBAR_CONFIG = {
        "SHOW_TOOLBAR_CALLBACK": show_toolbar,
        "SHOW_COLLAPSED": True,
    }
```

**Review by TNA-Allan — CHANGES_REQUESTED** [review]
Hi @ahosgood  -  A few things in the comments - as  I was not able to see the toolbar otherwise.

File: config/urls.py (hunk 1 above)
Comment (ahosgood) [reply config/urls.py:35]: This would be good. Please feel free to edit.

File: config/settings/develop.py (hunk 2 above)
Comment (ahosgood) [reply config/settings/develop.py:17]: That's all fine - please edit this PR

The only thing I'd say is that `"debug_toolbar.middleware.DebugToolbarMiddleware"` needs to be first on the list of `MIDDLEWARE`

**Review by TNA-Allan — APPROVED** [review]
Thanks @ahosgood

**Pull request merged**